src/hash-history-router.js

```diff
@@ -0,0 +1,45 @@
+// @flow
+import createHashHistory from 'history/lib/createHashHistory';
+import useBasename from 'history/lib/useBasename';
+import useQueries from 'history/lib/useQueries';
+
+import installRouter from './store-enhancer';
+import routerMiddleware from './middleware';
+
+type HashHistoryRouterArgs = {
+  routes: Object,
+  basename: string,
+  hashType: string,
+  getLocation: () => Location,
+  passRouterStateToReducer?: bool
+};
+
+/* istanbul ignore next: unstubbable! */
+const realLocation = () => window.location;
+
+export default ({
+  routes,
+  basename,
+  hashType = 'slash',
+  getLocation = realLocation,
+  passRouterStateToReducer = false
+}: HashHistoryRouterArgs) => {
+  const history = useBasename(useQueries(createHashHistory))({
+    basename,
+    hashType
+  });
+
+  const { pathname, search } = getLocation();
+  const location = history
+    .createLocation({ pathname, search });
```

Comment:
:warning: These parameters for `createLocation()` only work correctly for the BrowserRouter.

For this HashHistoryRouter, the `hash` property from `getLocation()` needs to be used.

I've made a fix here which seems to work correctly for me:

[**Fix bug: incorrect initial location when installing router**](https://github.com/michaelhogg/redux-little-router/commit/977964a)
```js
const { hash } = getLocation();

// The hash will begin with one of the following prefixes:
//     #     If hashType == 'noslash'
//     #/    If hashType == 'slash'
//     #!/   If hashType == 'hashbang'

const hashWithoutPrefix = hash.replace(/^((#\/)|(#!\/)|(#))/, '');

const pathname = `/${ hashWithoutPrefix }`;

const location = history.createLocation({ pathname });
```